modules/outbound-command-event-handler/src/domain/bulk_batch_entity.ts

```diff
@@ -0,0 +1,208 @@
+/*****
+ License
+ --------------
+ Copyright © 2017 Bill & Melinda Gates Foundation
+ The Mojaloop files are made available by the Bill & Melinda Gates Foundation under the Apache License, Version 2.0 (the "License") and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list (alphabetical ordering) of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+ * Modusbox
+ - Vijay Kumar Guthi <vijaya.guthi@modusbox.com>
+ --------------
+ ******/
+
+'use strict';
+
+import {
+    BaseEntityState,
+    BaseEntity,
+    SchemaValidationError,
+} from '@mojaloop/sdk-scheme-adapter-private-shared-lib';
+import { SDKSchemeAdapter, v1_1 as FSPIOP } from '@mojaloop/api-snippets';
+import { randomUUID } from 'crypto';
+import Ajv from 'ajv';
+import { BulkTransactionEntity } from './bulk_transaction_entity';
+const ajv = new Ajv({
+    strict:false,
+    allErrors: false,
+});
+
+// TODO: check name and status enums.
+export enum BulkBatchInternalState {
+    CREATED = 'CREATED',
+    AGREEMENT_PROCESSING = 'AGREEMENT_PROCESSING',
+    AGREEMENT_SUCCESS = 'AGREEMENT_SUCCESS',
+    AGREEMENT_FAILED = 'AGREEMENT_FAILED',
+    TRANSFER_PROCESSING = 'TRANSFER_PROCESSING',
+}
+
+export interface BulkBatchState extends BaseEntityState {
+    id: string;
+    state: BulkBatchInternalState;
+    bulkId: string;
+    bulkQuoteId: string;
+    bulkTransferId: string;
+    bulkQuotesRequest: SDKSchemeAdapter.Outbound.V2_0_0.Types.bulkQuoteRequest;
+    bulkQuotesResponse?: SDKSchemeAdapter.Outbound.V2_0_0.Types.bulkQuoteResponse;
+    bulkTransfersRequest: SDKSchemeAdapter.Outbound.V2_0_0.Types.bulkTransferRequest;
+    bulkTransfersResponse?: SDKSchemeAdapter.Outbound.V2_0_0.Types.bulkQuoteResponse;
+    quoteIdReferenceIdMap: {[quoteId: string]: string};
+    transferIdReferenceIdMap: {[quoteId: string]: string};
```

Comment:
```suggestion
    transferIdReferenceIdMap: {[transactionId: string]: string}; // Key = individual transactionId from request, Value = ?
```

If its not needed at the moment, comment it out with a comment indicating that it will be re-looked at as part of the transfers implementation story.